core/Plugin/Manager.php

```diff
@@ -1716,4 +1716,9 @@ private function savePluginTime($timingName, $pluginName)
         }
     }
 
+    public function getVersion($pluginName)
+    {
+        return $this->getLoadedPlugin($pluginName)->getVersion();
+    }
+
```

Comment:
This can currently cause an exception if a plugin was actually not loaded. This can e.g. happen if a plugin was installed but was removed on file system. Matomo then thinks the plugin should be there, as it is contained in the config, but then can't find it. Also plugins that are incompatible or deactivated might not be loaded in all cases.
I guess this code might be more accurate:

```suggestion
    public function getVersion(string $pluginName): ?string
    {
        if ($this->isPluginLoaded($pluginName)) {
            return (string) $this->getLoadedPlugin($pluginName)->getVersion();
        } elseif (!$this->isPluginInFilesystem($pluginName)) {
            return null;
        } else {
            try {
                $plugin = new Plugin($pluginName);
                return (string) $plugin->getVersion();
            } catch (\Exception $e) {
                return null;
            }
        }
    }
```

Reply:
makes sense!